src/com/tinyinswing/demo/Demo.java

```diff
@@ -0,0 +1,79 @@
+package com.tinyinswing.demo;
+
+import cloud.tiny.tinymceforswing.TinyMCE;
+import cloud.tiny.tinymceforswing.api.config.Config;
+import cloud.tiny.tinymceforswing.api.config.wrappers.ActionApi;
+import cloud.tiny.tinymceforswing.api.events.TinyListener;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.concurrent.ExecutionException;
+
+public class Demo extends JFrame {
+
+  public Demo() {
+    try {
+      init(this);
+    } catch (ExecutionException e) {
+      e.printStackTrace();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void init(JFrame locationRelativeTo) throws ExecutionException, InterruptedException {
+    setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+    setBounds(100, 100, 450, 450);
+
+    TinyListener insertLink = new TinyListener("javalink", new ActionApi() {
+      @Override
+      public void run(TinyMCE tinyMCE, String s) {
+        final JDialog popup = new JDialog();
+        popup.setTitle("Insert link");
+        popup.setSize(300, 60);
+        popup.setModal(true);
+        final JPanel panel = new JPanel(new FlowLayout());
+        final JTextField textfield = new JTextField();
```

Comment:
Specify `s` as a parameter:
 `new JTextField(s);`

Reply:
You could probably get rid of the `setPreferredSize` too if you set a column count.